include/graph.hpp

```diff
@@ -603,19 +603,20 @@ class graph {
     auto&
     make_node(Args&&... args) { // TODO for review: do we still need this factory method or allow only pmt-map-type constructors (see below)
         static_assert(std::is_same_v<Node, std::remove_reference_t<Node>>);
-        auto& new_node_ref = _nodes.emplace_back(std::make_unique<node_wrapper<Node>>(std::forward<Args>(args)...));
-        return *static_cast<Node*>(new_node_ref->raw());
+        auto &new_node_ref = _nodes.emplace_back(std::make_unique<node_wrapper<Node>>(std::forward<Args>(args)...));
+        auto                  raw_ref      = static_cast<Node *>(new_node_ref->raw());
+        [[maybe_unused]] auto _            = raw_ref->settings().apply_staged_parameters();
+        return *raw_ref;
     }
 
     template<typename Node>
     auto&
     make_node(const tag_t::map_type& initial_settings) {
         static_assert(std::is_same_v<Node, std::remove_reference_t<Node>>);
-        auto& new_node_ref = _nodes.emplace_back(std::make_unique<node_wrapper<Node>>());
-        auto raw_ref = static_cast<Node*>(new_node_ref->raw());
-        if (!initial_settings.empty()) {
-            static_cast<node<Node>*>(raw_ref)->settings().init(*raw_ref, initial_settings);
-        }
+        auto                 &new_node_ref = _nodes.emplace_back(std::make_unique<node_wrapper<Node>>());
+        auto                  raw_ref      = static_cast<Node *>(new_node_ref->raw());
+        [[maybe_unused]] auto _1           = raw_ref->settings().set(initial_settings);
```

Comment:
Maybe `std::ignore = ...` instead of` [[maybe_unused]] auto ...`?

Reply:
thanks @ivan-cukic for pointing out the use of `std::ignore` :+1: